### ISSUE TITLE
snapper cannot to be built with btrfs-progs v6.1

### ISSUE THREAD

**Issue by henrich** [openSUSE/snapper#762]
Hi,

I've got a bug report for Debian snapper package as [Debian Bug#1027628](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1027628), and I found that it was caused by a change in  `/usr/include/btrfs/kerncompat.h` in btrfs-progs v6.1 as https://github.com/kdave/btrfs-progs/commit/788a71c16a917f8357784571106537a1d42012e8

```shell
/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I..  -I/usr/include/libxml2   -Wdate-time -D_FORTIFY_SOURCE=2 -DCONF_DIR='"/etc/default"' -D_FILE_OFFSET_BITS=64 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -std=c++11 -Wall -Wextra -Wformat -Wnon-virtual-dtor -Wno-unused-parameter -Wsuggest-override -c -o libsnapper_la-MntTable.lo `test -f 'MntTable.cc' || echo './'`MntTable.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -I/usr/include/libxml2 -Wdate-time -D_FORTIFY_SOURCE=2 -DCONF_DIR=\"/etc/default\" -D_FILE_OFFSET_BITS=64 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -std=c++11 -Wall -Wextra -Wformat -Wnon-virtual-dtor -Wno-unused-parameter -Wsuggest-override -c MntTable.cc  -fPIC -DPIC -o .libs/libsnapper_la-MntTable.o
In file included from /usr/include/c++/12/bits/regex_compiler.h:571,
                 from /usr/include/c++/12/regex:65,
                 from Btrfs.cc:47:
/usr/include/c++/12/bits/regex_compiler.tcc: In member function ‘bool std::__detail::_Compiler<_TraitsT>::_M_quantifier()’:
/usr/include/c++/12/bits/regex_compiler.tcc:179:19: error: expected unqualified-id before ‘=’ token
  179 |       auto __init = [this, &__neg]()
      |                   ^
/usr/include/c++/12/bits/regex_compiler.tcc:187:18: error: expected primary-expression before ‘)’ token
  187 |           __init();
      |                  ^
/usr/include/c++/12/bits/regex_compiler.tcc:197:18: error: expected primary-expression before ‘)’ token
  197 |           __init();
      |                  ^
/usr/include/c++/12/bits/regex_compiler.tcc:205:18: error: expected primary-expression before ‘)’ token
  205 |           __init();
      |                  ^
make[3]: *** [Makefile:768: libsnapper_la-Btrfs.lo] Error 1
```

It's because `__init` was defined in `/usr/include/btrfs/kerncompat.h`

**Comment by aschnell** [comment]
Looks like a problem of btrfs-progs since names starting with __ are reserved by the implementation (compiler and sandard headers).

**Comment by aschnell** [comment]
As a workaround you can ```#undef __init``` in Btrfs.cc after including ```btrfs/send-utils.h```. I will contact the btrfs people anyway.

**Comment by aschnell** [comment]
Should be fixed with btrfs-progs 6.1.1, see https://lore.kernel.org/linux-btrfs/20230103173549.10791-1-dsterba@suse.com/T/#u.
